security/pkg/nodeagent/sds/server.go

```diff
@@ -205,8 +206,13 @@ func (s *Server) initGatewaySdsService(options *Options) error {
 			if err = s.grpcGatewayServer.Serve(s.grpcGatewayListener); err != nil {
 				log.Errorf("SDS grpc server for ingress gateway proxy failed to start: %v", err)
 			}
-			s.grpcGatewayListener, err = setUpUds(options.IngressGatewayUDSPath)
-			if err != nil {
+			time.Sleep(waitTime)
```

Comment:
1. if ```grpcGatewayServer.Serve``` fail, ```setUpUds``` should be called first(to clean/recreate uds path) before calling ```grpcGatewayServer.Serve``` again; so logic should be something like 
```
for i := 0; i < maxRetryTimes; i++ {
   if i != 0 {
       sleep(waittime)
   }
  ...grpcGatewayServer.Serve()
  ...setUpUds()
}
```
2. could you add something similar to ```initWorkloadSdsService``` ?

Reply:
Do we require a sequential order between grpcGatewayServer.Serve() and setUpUds(), or it doesn't matter? What will happen if setUpUds() fails and grpcGatewayServer.Serve() gets called twice?

Reply:
```setUpUds``` need to be called before ```Serve```

Reply:
I think this one should be fixed in a separate PR. I will investigate more and have another PR to fix this.

Reply:
better remove change in this function before check in